pennylane/transforms/zx/push_hadamards.py

```diff
@@ -0,0 +1,124 @@
+# limitations under the License.
+"""
+This module contains a transform to apply the
+`basic_optimization <https://pyzx.readthedocs.io/en/latest/api.html#pyzx.optimize.basic_optimization>`__
+pass (available through the external `pyzx <https://pyzx.readthedocs.io/en/latest/index.html>`__ package)
+to a PennyLane phase-polynomial + Hadamard circuit.
+"""
+
+from pennylane.tape import QuantumScript, QuantumScriptBatch
+from pennylane.transforms import transform
+from pennylane.typing import PostprocessingFn
+
+from .converter import from_zx, to_zx
+
+try:
+    import pyzx
+
+    has_pyzx = True
+except ModuleNotFoundError:
+    has_pyzx = False
+
+
+@transform
+def push_hadamards(tape: QuantumScript) -> tuple[QuantumScriptBatch, PostprocessingFn]:
+    """
+    Pushes Hadamard gates as far as possible to one side to cancel them and reduce the number of large phase-polynomial blocks,
+    improving the effectiveness of phase-polynomial optimization techniques.
+
+    This transform optimizes circuits composed of phase-polynomial blocks and Hadamard gates.
+    This strategy works by commuting Hadamard gates through the circuit.
+    To preserve the overall unitary, this process relies on commutation rules that can transform the gates a
+    Hadamard moves past. For instance, pushing a Hadamard through a CNOT gate will convert the latter into a
+    CZ gate. Consequently, the final optimized circuit may have, in some cases, a significantly different
+    internal gate structure.
+
+    The transform also applies some basic simplification rules to phase-polynomial blocks themselves to merge phase
+    gates together when possible (e.g. T^4 = S^2 = Z).
+
+    The implementation is based on the
+    `pyzx.basic_optimization <https://pyzx.readthedocs.io/en/latest/api.html#pyzx.optimize.basic_optimization>`__ pass.
+
+    Args:
+        tape (QNode or QuantumScript or Callable): the input circuit to be transformed.
+
+    Returns:
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumScript], function]:
+        the transformed circuit as described in :func:`qml.transform <pennylane.transform>`.
+
+    Raises:
+        ModuleNotFoundError: if the required ``pyzx`` package is not installed.
+        TypeError: if the input quantum circuit is not a phase-polynomial + Hadamard circuit.
+
+    **Example:**
+
+    .. code-block:: python3
+
+        import pennylane as qml
+        import pennylane.transforms.zx as zx
+
+        dev = qml.device("default.qubit")
+
+        @zx.push_hadamards
+        @qml.qnode(dev)
+        def circuit():
+            qml.T(0)
+            qml.Hadamard(0)
+            qml.Hadamard(0)
+            qml.T(1)
+            qml.Hadamard(1)
+            qml.CNOT([1, 2])
+            qml.Hadamard(1)
+            qml.Hadamard(2)
+            return qml.state()
+
+    ```pycon
+    >>> print(qml.draw(circuit)())
+    0: ──T────┤  State
+    1: ──T─╭X─┤  State
+    2: ──H─╰●─┤  State
+    ```
```

Comment:
```suggestion
.. code-block:: pycon

    >>> print(qml.draw(circuit)())
    0: ──T────┤  State
    1: ──T─╭X─┤  State
    2: ──H─╰●─┤  State

```
In the [rendered doc](https://xanaduai-pennylane--8025.com.readthedocs.build/en/8025/code/api/pennylane.transforms.zx.push_hadamards.html), this sample did not work well. Mind try this instead?
<img width="1208" height="820" alt="image" src="https://github.com/user-attachments/assets/65f2db5f-88fd-4e52-a60d-92a58bf60302" />